addons/binding/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/handler/HarmonyHubHandler.java

```diff
@@ -68,6 +69,16 @@
 
     public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections.singleton(HARMONY_HUB_THING_TYPE);
 
+    private static final Comparator<Activity> ACTIVITY_COMPERATOR = new Comparator<Activity>() {
+        @Override
+        public int compare(Activity a1, Activity a2) {
+            // if the order value is null we want it to be at the start of the list
+            int o1 = a1.getActivityOrder() == null ? -1 : a1.getActivityOrder().intValue();
+            int o2 = a2.getActivityOrder() == null ? -1 : a2.getActivityOrder().intValue();
+            return (o1 < o2) ? -1 : (o1 == o2) ? 0 : 1;
```

Comment:
? Would this simplify matters: https://stackoverflow.com/a/23908426/3656774

Reply:
Yes these Java 8 features certainly simplify the code! So it has been replaced.